src/android/ChromecastUtilities.java

```diff
@@ -0,0 +1,217 @@
+package acidhax.cordova.chromecast;
+
+import android.graphics.Color;
+import android.support.v4.content.ContextCompat;
+
+import com.google.android.gms.cast.MediaInfo;
+import com.google.android.gms.cast.MediaStatus;
+import com.google.android.gms.cast.MediaTrack;
+import com.google.android.gms.cast.TextTrackStyle;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.w3c.dom.Text;
+
+class ChromecastUtilities {
+
+    static String getMediaIdleReason(MediaStatus mediaStatus) {
+        switch(mediaStatus.getIdleReason()) {
+            case MediaStatus.IDLE_REASON_CANCELED:
+                return "canceled";
+            case MediaStatus.IDLE_REASON_ERROR:
+                return "error";
+            case MediaStatus.IDLE_REASON_FINISHED:
+                return "finished";
+            case MediaStatus.IDLE_REASON_INTERRUPTED:
+                return "interrupted";
+            case MediaStatus.IDLE_REASON_NONE:
+                return "none";
+            default:
+                return null;
+        }
+    }
+
+    static String getMediaPlayerState(MediaStatus mediaStatus) {
+        switch(mediaStatus.getPlayerState()) {
+            case MediaStatus.PLAYER_STATE_BUFFERING:
+                return "BUFFERING";
+            case MediaStatus.PLAYER_STATE_IDLE:
+                return "IDLE";
+            case MediaStatus.PLAYER_STATE_PAUSED:
+                return "PAUSED";
+            case MediaStatus.PLAYER_STATE_PLAYING:
+                return "PLAYING";
+            case MediaStatus.PLAYER_STATE_UNKNOWN:
+                return "UNKNOWN";
+            default:
+                return null;
+        }
+    }
+
+    static String getMediaInfoStreamType(MediaInfo mediaInfo) {
+        switch(mediaInfo.getStreamType()) {
+            case MediaInfo.STREAM_TYPE_BUFFERED:
+                return "buffered";
+            case MediaInfo.STREAM_TYPE_LIVE:
+                return "live";
```

Comment:
Are any of these values enums that could be parsed with a call to valueOf() or did they stick with integers?

Reply:
No, they can't. Because the name of the constant in JS is different to the name of the enum value in Java.

E.G. "STREAM_TYPE_LIVE" != "live"

Reply:
Technically we could use strings that match these values if they were enums, such as passing `STREAM_TYPE_LIVE` instead of `live` to the plugin. I think the values are flags though, so this won't work anyway.